src/contents/ui/Compact.qml

```diff
@@ -51,7 +53,43 @@ Item {
     Rectangle {
         anchors.fill: parent
         color: backgroundColor
-        radius: plasmoid.configuration.panelBackgroundRadius
+        radius: compact.panelBackgroundRadius
+        Item {
+            // start the progress after the cover/thumbnail if it is as tall as the panel thickness
+            property int panelIconSpace: panelIcon.type === PanelIcon.Type.Image && (compact.iconSize === compact.widgetThickness) ? compact.iconSize - Math.min(plasmoid.configuration.albumCoverRadius, compact.iconSize / 2) : 0
+            x: horizontal ? panelIconSpace : 0
+            y: horizontal ? 0 : panelIconSpace
+            width: horizontal ? parent.width - panelIconSpace : parent.width
+            height: horizontal ? parent.height : parent.height - panelIconSpace
+            Rectangle {
+                id: progress
+                color: foregroundColor
+                height: horizontal ? parent.height : parent.height * (player.songPosition / player.songLength)
+                width: horizontal ? parent.width * (player.songPosition / player.songLength) : parent.width
+                visible: plasmoid.configuration.mediaProgressInPanel
+                opacity: player.playbackStatus === Mpris.PlaybackStatus.Playing ? 0.15 : 0.07
+            }
+            Rectangle {
+                height: horizontal ? progress.height : 1
+                width: horizontal ? 1 : progress.width
+                color: foregroundColor
+                anchors.right: horizontal ? progress.right : undefined
+                anchors.bottom: horizontal ? undefined : progress.bottom
+                visible: progress.visible
+                opacity: 0.2
+            }
```

Comment:
I prefer the version without this bar. With the bar enabled, the progress animation feels snappy, whereas it appears smooth without it.